File: WebSpider_src/Mf/MfServiceVb/Vb/Service/WebSpider/WebPageState.vb
Imports System
Imports System.Net

Imports Mf.Util

Namespace Mf.Service.WebSpider

   Public Class WebPageState

      Private Sub New()
      End Sub

      Public Sub New(ByVal uri As Uri)
         m_uri = uri
      End Sub

      Public Sub New(ByVal uri As String)
         MyClass.New(New Uri(uri))
      End Sub

#Region "properties"
      Private m_uri As Uri
      Private m_content As String
      Private m_processInstructions As String = ""
      Private m_processStarted As Boolean = False
      Private m_processSuccessfull As Boolean = False
      Private m_statusCode As String
      Private m_statusDescription As String

      Public ReadOnly Property Uri() As Uri
         Get
            Return m_uri
         End Get
      End Property

      Public Property ProcessStarted() As Boolean
         Get
            Return m_processStarted
         End Get
         Set(ByVal Value As Boolean)
            m_processStarted = Value
         End Set
      End Property
      Public Property ProcessSuccessfull() As Boolean
         Get
            Return m_processSuccessfull
         End Get
         Set(ByVal Value As Boolean)
            m_processSuccessfull = Value
         End Set
      End Property

      Public Property ProcessInstructions() As String
         Get
            Return m_processInstructions
         End Get
         Set(ByVal Value As String)
            m_processInstructions = Value
         End Set
      End Property

      Public Property Content() As String
         Get
            Return m_content
         End Get
         Set(ByVal Value As String)
            m_content = Value
         End Set
      End Property

      Public Property StatusCode() As String
         Get
            Return m_statusCode
         End Get
         Set(ByVal Value As String)
            m_statusCode = Value
         End Set
      End Property

      Public Property StatusDescription() As String
         Get
            Return m_statusDescription
         End Get
         Set(ByVal Value As String)
            m_statusDescription = Value
         End Set
      End Property
#End Region

   End Class

End Namespace

File: WebSpider_src/Mf/MfService/Service/WebSpider/WebPageState.cs
using System;
using System.Net;

using Mf.Util;

namespace Mf.Service.WebSpider
{
	/// <summary>
	/// Summary description for WebPage.
	/// </summary>
	public class WebPageState
	{
      private WebPageState( ) {}

		public WebPageState( Uri uri )
		{
         m_uri             = uri;
		}

      public WebPageState( string uri ) 
         : this( new Uri( uri ) ) { }

      #region properties
      Uri            m_uri;
      string         m_content;
      string         m_processInstructions   = "";
      bool           m_processStarted        = false;
      bool           m_processSuccessfull    = false;
      string         m_statusCode;
      string         m_statusDescription;

      public Uri Uri
      {
         get
         {
            return m_uri;
         }
      }
      public string Content
      {
         get
         {
            return m_content;
         }
         set
         {
            m_content = value;
         }
      }
      public bool ProcessStarted
      {
         get
         {
            return m_processStarted;
         }
         set
         {
            m_processStarted = value;
         }
      }
      public bool ProcessSuccessfull
      {
         get
         {
            return m_processSuccessfull;
         }
         set
         {
            m_processSuccessfull = value;
         }
      }
      public string ProcessInstructions
      {
         get
         {
            return ( m_processInstructions == null ? "" : m_processInstructions );
         }
         set
         {
            m_processInstructions = value;
         }
      }
      public string StatusCode
      {
         get
         {
            return m_statusCode;
         }
         set
         {
            m_statusCode = value;
         }
      }
      public string StatusDescription
      {
         get
         {
            return m_statusDescription;
         }
         set
         {
            m_statusDescription = value;
         }
      }
      #endregion
	}
}
